core/lib/eth_signer/src/raw_ethereum_tx.rs

```diff
@@ -147,6 +155,42 @@ impl Transaction {
         stream
     }
 
+    /// Encodes a [`eip4844`] transaction: <https://eips.ethereum.org/EIPS/eip-4844>
+    fn encode_eip4844_payload(&self, chain_id: u64, signature: Option<&Signature>) -> RlpStream {
```

Comment:
do we have any unittests for this? if not, I'd suggest adding at least one in this file

Reply:
this PR contains unittests.